autoload/maktaba/syscall.vim

```diff
@@ -135,7 +135,14 @@ endfunction
 function! maktaba#syscall#DoCallForeground(pause) abort dict
   let l:return_data = {}
   if a:pause
-    execute '!' . s:EscapeSpecialChars(self.GetCommand())
+    if has('nvim')
+        let l:cmd = 'noautocmd new | terminal '
+    elseif has('terminal')
+        let l:cmd = 'terminal '
+    else
+        let l:cmd = '!'
+    endif
+    execute l:cmd . s:EscapeSpecialChars(self.GetCommand())
```

Comment:
Could you define a `maktaba#syscall#SetTermCmdDisabledForTesting` helper to bypass this where needed for testing? IIUC it's going to have slightly different usage behavior so that any vroom scripts that try to run a command and press ENTER to clear it would have to get smarter per configuration.